apps/aehyperchains/test/aehc_contract_tests.erl

```diff
@@ -0,0 +1,675 @@
+    Contract = aect_contracts:compute_contract_pubkey(Restricted, Nonce),
+    aec_chain_sim:dict_set(staking_contract, Contract), staking_contract().
+
+
+call_staking_contract(Fun, Args, Type) ->
+    call_staking_contract(restricted_account(), Fun, Args, Type).
+call_staking_contract(Acct, Fun, Args, Type) ->
+    call_staking_contract(Acct, 0, Fun, Args, Type).
+call_staking_contract(Acct, Value, Fun, Args, Type) ->
+    CallData = make_calldata(Fun, Args),
+    Nonce = aec_chain_sim:next_nonce(Acct),
+    {ok, Tx} = aect_call_tx:new(
+               #{ caller_id   => aeser_id:create(account, Acct)
+               ,  nonce       => Nonce
+               ,  contract_id => aeser_id:create(contract, staking_contract())
+               ,  abi_version => ?ABI_FATE_SOPHIA_1
+               ,  fee         => ?CALL_COST
+               ,  amount      => Value
+               ,  gas         => ?CALL_GAS
+               ,  gas_price   => aec_test_utils:min_gas_price()
+               ,  call_data   => CallData
+               }),
+    aec_chain_sim:sign_and_push(Acct, Tx),
+    add_microblock(),
+    CallId = aect_call:id(Acct, Nonce, staking_contract()),
+    {Result, Gas} = get_result(Type, CallId),
+    Restricted = restricted_account(),
+    {ok, PayBackTx} = aec_spend_tx:new(
+        #{ sender_id    => aeser_id:create(account, Restricted)
+        ,  recipient_id => aeser_id:create(account, Acct)
+        ,  amount       => Gas * aec_test_utils:min_gas_price() + ?CALL_COST
+        ,  fee          => ?CALL_COST
+        ,  nonce        => aec_chain_sim:next_nonce(Restricted)
+        ,  payload      => <<"payback">>
+        }),
+    aec_chain_sim:sign_and_push(Restricted, PayBackTx),
+    add_microblock(),
+    Result.
+
+get_result(Type, CallId) ->
+    Call = aec_chain_sim:get_call(staking_contract(), CallId),
+    Result = case aect_call:return_type(Call) of
+                 ok ->
+                     Res = aeb_fate_encoding:deserialize(aect_call:return_value(Call)),
+                     case aefate_test_utils:decode(Res, Type) of
+                        {variant, [0,1], 0, {}} when element(1, Type) =:= option ->
+                            none;
+                        {variant, [0,1], 1, {Decoded}} when element(1, Type) =:= option ->
+                            {some, Decoded};
+                        Decoded ->
+                            Decoded
+                     end;
+                 error ->
+                     {error, aect_call:return_value(Call)};
+                 revert ->
+                     Res = aeb_fate_encoding:deserialize(aect_call:return_value(Call)),
+                     {revert, aefate_test_utils:decode(Res)}
+             end,
+    {Result, aect_call:gas_used(Call)}.
+
+fake_random() ->  %% We want to make it predictable
+    <<1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32>>.
+
+%%
+%% ENTRYPOINTS
+%%
+-define(assertSuccess(X), (fun () -> case X of
```

Comment:
Use macros from Eunit :P

Reply:
You mean `assertMatch`? Wasn't sure if it will return the resulting value